tests/integration_tests/test_source_selection.py

```
@@ -0,0 +1,304 @@
+        """Test: Extracting array field from metadata"""
+        result = collection.get(ids="doc_1", _source=["metadata.tags"])
+
+        meta = result["metadatas"][0]
+        assert "tags" in meta
+        assert meta["tags"] == ["AI", "Database", "Vector"]
+
+    # ==================== Type Validation Tests ====================
+
+    def test_source_invalid_type_string(self, collection):
+        """Test: _source must be list, not string"""
+        with pytest.raises(Exception):
```

Ruff (0.14.14)

Do not assert blind exception: `Exception`

(B017)